File: include/extras-kwidgetsaddons/libkmimetypeeditor.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");

/// https://api-staging.kde.org/kmimetypeeditor.html
pub const kmimetypeeditor = struct {
    /// [Qt documentation](https://api-staging.kde.org/kmimetypeeditor.html#editMimeType)
    ///
    /// ``` param1: []const u8, param2: QtC.QWidget ```
    pub fn EditMimeType(param1: []const u8, param2: ?*anyopaque) void {
        const param1_str = qtc.libqt_string{
            .len = param1.len,
            .data = param1.ptr,
        };
        qtc.KMimeTypeEditor_EditMimeType(param1_str, @ptrCast(param2));
    }
};

File: src_original/extras-kwidgetsaddons/libkmimetypeeditor.cpp
#include <KMimeTypeEditor>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QWidget>
#include <kmimetypeeditor.h>
#include "libkmimetypeeditor.h"
#include "libkmimetypeeditor.hxx"

void KMimeTypeEditor_EditMimeType(const libqt_string param1, QWidget* param2) {
    QString param1_QString = QString::fromUtf8(param1.data, param1.len);
    KMimeTypeEditor::editMimeType(param1_QString, param2);
}
